utils.go

```diff
@@ -0,0 +1,18 @@
+package nmap
+
+// https://siongui.github.io/2018/04/14/go-remove-duplicates-from-slice-or-array/
+func RemoveDuplicatesFromSlice(s []string) []string {
+	m := make(map[string]bool)
```

Comment:
For such uses, prefer an empty struct to a `bool`, since here there's no differentiation between `true` and `false`, there just is or isn't something, so an empty struct is a better representation of the state in this case

[More info here](https://dave.cheney.net/2014/03/25/the-empty-struct)

```suggestion
	m := make(map[string]struct{})
```